src/app/cli/src/graphql.ml

```diff
@@ -564,19 +564,37 @@ module Types = struct
           | _ ->
               Error "Invalid format for public key." )
 
-    let uint64_arg =
-      scalar "UInt64" ~doc:"String representing a uint64 number in base 10"
+    module type Numeric_type = sig
+      type t
+
+      val of_string : string -> t
+
+      val of_int : int -> t
+    end
+
+    (** Converts a type into a graphql argument type. Expect name to start with uppercase    *)
+    let make_numeric_arg (type t) ~name
+        (module Numeric : Numeric_type with type t = t) =
+      let lower_name = String.lowercase name in
+      scalar name
+        ~doc:(sprintf "String representing a %s number in base 10" lower_name)
         ~coerce:(fun key ->
           match key with
           | `String s ->
-              result_of_exn Unsigned.UInt64.of_string s
-                ~error:"Could not decode uint64."
+              result_of_exn Numeric.of_string s
+                ~error:(sprintf "Could not decode %s." lower_name)
           | `Int n ->
               if n < 0 then
-                Error "Could not convert negative number to uint64."
-              else Ok (Unsigned.UInt64.of_int n)
+                Error
```

Comment:
If the documentation says string then we shouldn't accept ints.

Reply:
Seems like there's no other way to transport numbers larger than `int` through this GraphQL interface...